### PR TITLE
improvements to handle refspec and result comparision

### PR DESCRIPTION
**Pull request by amarts**
* git fetch and then checkout to FETCH_HEAD, handle refspec (in prereq)
* git checkout to master, and delete branch in cleanup
* break resdir into resdir_base, and resdir, which helps in organising
  the results per volume.
* add another script which is only to compare the results, and
  give text output, instead of graphs. use the same in sendmail.
* add a default 'tag' of nightly.

### CODE REVIEW

**Comment by amarts** [comment]
this is similar to #33, but contains changes which are minimal.

**Comment by amarts** [comment]
@aravindavk check if git section is valid now.

File: playbooks/cleanup.yml [hunk 1]
```diff
@@ -50,3 +50,13 @@
...
        ignore_errors: true
        with_items: "{{ gluster_infra_mount_devices }}"
+
+     - name: Checkout to Master
+       command: git checkout master
+       args:
+         chdir: "{{ glusterfs_perf_git_dest | default('/usr/src/glusterfs') }}"
+
+     - name: Delete Branch
+       command: for br in $(git branch | grep test); do git branch -D $br; done
```
Comment (aravindavk) [inline playbooks/cleanup.yml]: pipe and other things may not work when used `command:`, use `shell:`

Comment (amarts) [reply playbooks/cleanup.yml]: Ack! Will change.

**Comment by amarts** [comment]
@sac @aravindavk any further comments?

**Comment by sac** [comment]
> @sac @aravindavk any further comments?

Now if the to list is not defined it'll not send mail. But what about sending single email?

**Comment by amarts** [comment]
Will work on it as a separate patch. This is just a groundwork, and currently this doesn't change anything in current behavior of sending mails.

**Pull request merged**